BasicCrudOperations/Controllers/CourseController.cs

```diff
@@ -15,34 +16,60 @@ public CourseController(ICourseService courseService)
             _courseService = courseService;
         }
 
-        [HttpGet("GetCourses")]
+        [HttpGet("Courses")]
         public IActionResult Get()
         {
-            return Ok(_courseService.GetCourses());
+            var courses = _courseService.GetCourses();
+            if (courses == null)
+            {
+                return NotFound();
+            }
+            return Ok(courses);
         }
 
-        [HttpGet("GetCourseByID")]
-        public IActionResult Get([FromQuery] int id)
+        [HttpGet("Courses/{id}")]
+        public IActionResult Get([FromRoute] int id)
         {
-            return Ok(_courseService.GetCourseByID(id));
+            var course = _courseService.GetCourseByID(id);
+            if (course == null)
+            {
+                return NotFound();
+            }
+            return Ok(course);
         }
 
-        [HttpPost("AddCourse")]
-        public IActionResult Post([FromBody] CourseDTO course)
+        [HttpPost("Courses")]
+        public IActionResult Post([FromBody] CourseDTO courseDTO)
         {
-            return Ok(_courseService.AddCourse(course));
+            var course = _courseService.AddCourse(courseDTO);
+            if (course == null)
+            {
+                return NotFound();
```

Comment:
When adding a new entity, you can't return `NotFound`, you are not searching for it, if sent parameters aren't valid, that should be handled using validations inside `CourseDTO`, but if some error occurs in repo while adding new entity, that should be `ServerError - 500 status code`